### PR TITLE
requestlog: add traceID

### PR DESCRIPTION
**Pull request by felipeweb**
ref: https://github.com/google/go-cloud/issues/1581

### CODE REVIEW

File: requestlog/stackdriver.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ func (l *StackdriverLogger) log(ent *Entry) error {
 			Seconds int64 `json:"seconds"`
 			Nanos   int   `json:"nanos"`
 		} `json:"timestamp"`
+		TraceID string `json:"trace"`
```
Comment (zombiezen) [inline requestlog/stackdriver.go]: Unfortunately, this isn't a `LogEntry` object, this is the JSON data structure parsed by the [Fluent plugin](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/out_google_cloud.rb), so the `trace` key isn't directly recognized. (Would you mind adding a comment to that effect somewhere? This isn't an obvious property of this code.)

Glancing at the plugin's source, it appears that `logging.googleapis.com/trace` and `logging.googleapis.com/spanId` are the two keys we're interested in ([source](https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/f93046d92f7722db2794a042c3f2dde5df91a90b/lib/fluent/plugin/out_google_cloud.rb#L154-L155)).

File: requestlog/requestlog.go [hunk 2]
```diff
@@ -107,6 +112,7 @@ type Entry struct {
 	ResponseHeaderSize int64
 	ResponseBodySize   int64
 	Latency            time.Duration
+	TraceID            string
 }
```
Comment (zombiezen) [inline requestlog/requestlog.go]: We also should capture the [span ID](https://opencensus.io/tracing/span/). (Similar to the above comment, let's use the `trace.SpanID` type for this field.)

File: requestlog/requestlog.go [hunk 3]
```diff
@@ -107,6 +112,7 @@ type Entry struct {
 	ResponseHeaderSize int64
 	ResponseBodySize   int64
 	Latency            time.Duration
+	TraceID            string
```
Comment (zombiezen) [inline requestlog/requestlog.go]: Since this is our abstract representation of a request log entry, we are permitted to have more strong types here. Let's use OpenCensus's `TraceID` type directly to make it clear what format it is in.

```suggestion
	TraceID            trace.TraceID
```

File: requestlog/requestlog_test.go [hunk 4]
```diff
@@ -75,6 +80,9 @@ func TestHandler(t *testing.T) {
 	if ent.ResponseBodySize != int64(len(responseMsg)) {
 		t.Errorf("ResponseBodySize = %d; want %d", ent.ResponseBodySize, len(responseMsg))
 	}
+	if ent.TraceID == "" {
```
Comment (zombiezen) [inline requestlog/requestlog_test.go]: I think we have enough information to verify the trace ID matches the one in the span, right? IIUC that would give us more confidence in this test than just testing for non-empty.

**Comment by felipeweb** [comment]
@zombiezen done! Let me know if it needs something more.

File: internal/testing/alldeps [hunk 5]
```diff
@@ -37,6 +37,7 @@ github.com/hashicorp/go-multierror
 github.com/hashicorp/go-retryablehttp
 github.com/hashicorp/go-rootcerts
 github.com/hashicorp/go-sockaddr
+github.com/hashicorp/golang-lru
```
Comment (zombiezen) [inline internal/testing/alldeps]: To anyone else auditing here: this dependency is released under Mozilla Public License 2.0, which we already have cleared to use from the MySQL driver.

File: requestlog/requestlog_test.go [hunk 6]
```diff
@@ -75,20 +76,41 @@ func TestHandler(t *testing.T) {
...
+	ctx, span := trace.StartSpan(r.Context(), "test")
+	defer span.End()
+	r = r.WithContext(ctx)
+	sc := trace.FromContext(ctx).SpanContext()
+	sh.spanCtx = &sc
+	sh.h.ServeHTTP(w, r)
 }
 
-func roundTrip(r *http.Request, h http.Handler) (*Entry, error) {
+func roundTrip(r *http.Request, h http.Handler) (*Entry, trace.SpanContext, error) {
```
Comment (zombiezen) [inline requestlog/requestlog_test.go]: nit: Seems like this should return a `*trace.SpanContext`?

Comment (felipeweb) [reply requestlog/requestlog_test.go]: @zombiezen https://github.com/google/go-cloud/pull/1628/commits/e8907ea7eba02b57e4a276f25619f27b534b1a58
